File: TransCoder_tokenized_test_set_functions/cpp/FIND_REPEATED_CHARACTER_PRESENT_FIRST_STRING_1.cpp
int findRepeatFirst ( char * s ) {
  int p = - 1, i, k;
  int hash [ MAX_CHAR ] = {
    0 };
    int pos [ MAX_CHAR ];
    for ( i = 0;
    i < strlen ( s );
    i ++ ) {
      k = ( int ) s [ i ];
      if ( hash [ k ] == 0 ) {
        hash [ k ] ++;
        pos [ k ] = i;
      }
      else if ( hash [ k ] == 1 ) hash [ k ] ++;
    }
    for ( i = 0;
    i < MAX_CHAR;
    i ++ ) {
      if ( hash [ i ] == 2 ) {
        if ( p == - 1 ) p = pos [ i ];
        else if ( p > pos [ i ] ) p = pos [ i ];
      }
    }
    return p;
  }

File: TransCoder_tokenized_test_set_functions/python/FIND_REPEATED_CHARACTER_PRESENT_FIRST_STRING_1.py
def findRepeatFirst ( s ) :
    p = - 1
    hash = [ 0 for i in range ( MAX_CHAR ) ]
    pos = [ 0 for i in range ( MAX_CHAR ) ]
    for i in range ( len ( s ) ) :
        k = ord ( s [ i ] )
        if ( hash [ k ] == 0 ) :
            hash [ k ] += 1
            pos [ k ] = i
        elif ( hash [ k ] == 1 ) :
            hash [ k ] += 1
    for i in range ( MAX_CHAR ) :
        if ( hash [ i ] == 2 ) :
            if ( p == - 1 ) :
                p = pos [ i ]
            elif ( p > pos [ i ] ) :
                p = pos [ i ]
    return p
